src/app/components/pause/pause.component.ts

```diff
@@ -1,26 +1,50 @@
-import { AsyncPipe } from '@angular/common';
-import { Component } from '@angular/core';
 import { GameState } from '@angular-tetris/interface/game-state';
-import { TetrisQuery } from '@angular-tetris/state/tetris/tetris.query';
-import { interval, Observable, of } from 'rxjs';
-import { map, switchMap } from 'rxjs/operators';
+import { TetrisStateService } from '@angular-tetris/state/tetris/tetris.state';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  DestroyRef,
+  effect,
+  inject,
+  signal
+} from '@angular/core';
 
 @Component({
   selector: 't-pause',
   standalone: true,
-  imports: [AsyncPipe],
   templateUrl: './pause.component.html',
-  styleUrls: ['./pause.component.scss']
+  styleUrls: ['./pause.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class PauseComponent {
-  paused$: Observable<boolean> = this.query.gameState$.pipe(
-    switchMap((state) => {
-      if (state === GameState.Paused) {
-        return interval(250).pipe(map((num) => !!(num % 2)));
-      }
-      return of(false);
-    })
-  );
```

Comment:
Let's stick with the old RxJS implementation here as well, looks so much easy to understand.